src/actions/lender.ts

```diff
@@ -39,6 +56,18 @@ export function LenderActions<ActionBase extends Constructor<TinlakeParams>>(Bas
       return this.pending(this.contract('JUNIOR_OPERATOR').supplyOrder(supplyAmount, this.overrides))
     }
 
+    submitJuniorSupplyOrderWithPermit = async(amount: string, senderAddress: string) => {
+      if (!this.contractAddresses['JUNIOR_TOKEN'] || !this.contractAddresses['JUNIOR_TRANCHE']) {return}
```

Comment:
This also needs to support DAI. We should probably create a generic function to avoid code duplication and then pass in the tranche address

Reply:
we have code duplication in almost all the other junior/senior methods, to avoid having to pass in all the addresses I guess. I think being more explicit with the function name makes more sense in this case.